app/models/auth_key.rb

```diff
@@ -41,8 +41,8 @@ def generate_unique_key
     end
   end
 
-  def touch_device(token)
-    Device.touch_device(self, token)
+  def touch_device(token, label = nil)
+    Device.touch_device(self, token, label)
```

Comment:
You can just pass `options` through to the class method if you implement the other suggestion.